jgrapht-core/src/main/java/org/jgrapht/traverse/LexBreadthFirstIterator.java

```diff
@@ -387,48 +428,91 @@ V poll()
-                    bucketMap.put(vertex, bucket.prev);
-                } else {
-                    visitedBuckets.add(bucket);
-                    Bucket newBucket;
-                    if (priorityB != null) {
-                        newBucket = new Bucket(vertex, new PriorityComparator(priorityA), new PriorityComparator(priorityB));
-                    }
-                    else{
-                        newBucket = new Bucket(vertex, priorityComparator);
+        void updateBuckets(V vertex) {
+            HashSet<Bucket> bucketsToUpdate = new HashSet<>();
+            Map<Bucket, List<V>> intersectionLists = new HashMap<Bucket, List<V>>();
+            Map<Bucket, List<V>> intersectionListsB = null;
+            
+            List<V> sortedNeighborhood = sortedNeighbors.get(vertex);
+            List<V> sortedNeighborhoodB = null;
+            
+            for(V neighbor : sortedNeighborhood) {
+                Bucket B = bucketMap.get(neighbor);
+                
+                // If this vertex was already traversed, do not consider it
+                if(B == null) {
+                    continue;
+                }
+                
+                // If necessary, memorize that we need to update B and initialize a sorted list which holds the intersection of B and N(vertex)
+                if(!bucketsToUpdate.contains(B)) {
+                    bucketsToUpdate.add(bucketMap.get(neighbor));
+                    intersectionLists.put(B, new LinkedList<V>());
+                }
+                
+                // Add neighbor to said intersection list
+                intersectionLists.get(B).add(neighbor);
+            }
+            
+            if(sortedNeighborsB != null) {
+                sortedNeighborhoodB = sortedNeighborsB.get(vertex);
+                intersectionListsB = new HashMap<Bucket, List<V>>();
+                
+                for(V neighbor : sortedNeighborhoodB) {
+                    Bucket B = bucketMap.get(neighbor);
+                    
+                    // If this vertex was already traversed, do not consider it
+                    if(B == null) {
+                        continue;
                     }
-                    newBucket.insertBefore(bucket);
-                    bucketMap.put(vertex, newBucket);
-                    if (head == bucket) {
-                        head = newBucket;
+                    
+                    if(!intersectionListsB.containsKey(B)) {
+                        intersectionListsB.put(B, new LinkedList<V>());
                     }
+                    
+                    // Add neighbor to intersection list B
+                    intersectionListsB.get(B).add(neighbor);
+                }
+            }
+            
+            for(Bucket bucket : bucketsToUpdate) {
+                Bucket newBucket;
+                
+                // Split the bucket...
+                if(intersectionListsB == null) {
```

Comment:
maybe explain the meaning of this distinction (that null means not LexBFS*)

Reply:
done